app/src/main/java/grinnell/appdev/edu/lyles/ItemAdapter.java

```diff
@@ -0,0 +1,160 @@
+    private boolean mIsFavoritesTabClicked;
+
+    private int expandedIndex = -1;
+    private RecyclerView mRecyclerView;
+
+    public ItemAdapter(Context context, ArrayList<MenuItem> menuItems, boolean favTab) {
+        mContext = context;
+        mMenuItems = menuItems;
+        mFavoritesManager = new FavoritesManager(context, menuItems);
+        mIsFavoritesTabClicked = favTab;
+    }
+
+    @Override
+    public ItemAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+        mRecyclerView = (RecyclerView) parent;
+
+        View menuItemView = inflater.inflate(R.layout.item_user_card, parent, false);
+
+        ViewHolder viewHolder = new ViewHolder(menuItemView);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(final ViewHolder viewHolder, int position) {
+
+        final MenuItem menuItem = mMenuItems.get(position);
+        final int itemPosition = position;
+
+        TextView titleTextView = viewHolder.titleTextView;
+        titleTextView.setText(menuItem.getTitle());
+
+        TextView priceTextView = viewHolder.priceTextView;
+        priceTextView.setText(DOLLAR_SIGN + menuItem.getPrice());
+
+        ImageView imageView = viewHolder.imageView;
+        Glide.with(this.getContext()).load(menuItem.getImageUrl())
+                .override(300, 300).into(imageView);
+
+        ImageButton favoriteButton = viewHolder.favoriteButton;
+        setFaveButtonDrawable(viewHolder);
+        favoriteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                mFavoritesManager.toggleFavorite(menuItem.getTitle());
+                setFaveButtonDrawable(viewHolder);
+                if (mIsFavoritesTabClicked) {
+                    mMenuItems.remove(menuItem);
+                    notifyItemRemoved(itemPosition);
+                }
+            }
+        });
+
+        TextView detailsTextView = viewHolder.detailsTextView;
+        detailsTextView.setText(menuItem.getDetails());
+    }
+
+    @Override
+    public int getItemCount() {
+        return mMenuItems.size();
+    }
+
+    private Context getContext() {
+        return mContext;
+    }
+
+    /**
+     * Changes a view to or from its expanded mode
+     *
+     * @param  viewHolder  the ViewHolder to be altered
+     */
+    private void expandContractItem(ViewHolder viewHolder) {
+        if (viewHolder.detailsTextView.getVisibility() == View.GONE) {
+            viewHolder.detailsTextView.setVisibility(View.VISIBLE);
+        }
+        else {
+            viewHolder.detailsTextView.setVisibility(View.GONE);
+        }
+        ObjectAnimator animator = ObjectAnimator.ofInt(viewHolder.detailsTextView, "maxLines", TextViewCompat.getMaxLines(viewHolder.detailsTextView));
+        animator.setDuration(500).start();
```

Comment:
Extract 500 as its own constant as: `private final int MS_DURATION_ANIMATOR = 500` where the MS prefix indicates that it is in milliseconds